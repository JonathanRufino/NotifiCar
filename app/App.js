import React, { Component } from 'react';
import { Provider } from 'react-redux';
import { Platform, AsyncStorage, AppState, Alert } from 'react-native';
import FCM, {
    FCMEvent, 
    RemoteNotificationResult,
    WillPresentNotificationResult, 
    NotificationType,
} from 'react-native-fcm';
import { Actions } from 'react-native-router-flux';

import Routes from './routes/index';
import reduxStore from './redux/store';

registerKilledListener();

class App extends Component {
    componentDidMount() {
        FCM.requestPermissions()
            .then(() => {
                console.log('notification permission garanted');
                registerAppListener();
            })
            .catch(() => console.log('notification permission rejected'));
    }

    render() {
        return (
            <Provider store={reduxStore}>
                <Routes /> 
            </Provider>
        );
    }
}

export default App;

// these callback will be triggered even when app is killed
function registerKilledListener() {
    FCM.on(FCMEvent.Notification, notif => {
        AsyncStorage.setItem('lastNotification', JSON.stringify(notif));

        FCM.presentLocalNotification({
            title: Platform.OS === 'ios' ? notif.aps.alert.title : notif.fcm.title,
            body: Platform.OS === 'ios' ? notif.aps.alert.body : notif.fcm.body,
            sound: 'default',
            priority: 'high',
            vibrate: 300,
            wake_screen: true,
            show_in_foreground: true,
            click_action: 'com.jldevs.notificar', // for ios
        });

        if (notif.opened_from_tray) {
            setTimeout(() => {
                if (notif._actionIdentifier === 'reply') {
                    if (AppState.currentState !== 'background') {
                        // Alert.alert(`User replied ${JSON.stringify(notif._userText)}`);
                        Alert.alert('User replied');
                    } else {
                        AsyncStorage.setItem('lastMessage', JSON.stringify(notif._userText));
                    }
                }
                if (notif._actionIdentifier === 'view') {
                    Alert.alert('User clicked View in App');
                }
                if (notif._actionIdentifier === 'dismiss') {
                    Alert.alert('User clicked Dismiss');
                }
            }, 1000);
        }
    });
}

// these callback will be triggered only when app is foreground or background
function registerAppListener() {
    FCM.on(FCMEvent.Notification, notif => {
        FCM.presentLocalNotification({
            title: Platform.OS === 'ios' ? notif.aps.alert.title : notif.fcm.title,
            body: Platform.OS === 'ios' ? notif.aps.alert.body : notif.fcm.body,
            sound: 'default',
            priority: 'high',
            vibrate: 300,
            wake_screen: true,
            show_in_foreground: true,
            click_action: 'com.jldevs.notificar', // for ios
        });

        if (Platform.OS === 'ios'
                && notif._notificationType === NotificationType.WillPresent
                && !notif.local_notification) {
            // this notification is only to decide if you want to 
            // show the notification when user if in foreground.
            // usually you can ignore it. just decide to show or not.
            notif.finish(WillPresentNotificationResult.All);
            return;
        }
  
        if (notif.opened_from_tray) {
            if (notif.targetScreen === 'detail') {
                setTimeout(() => {
                    Actions.login();
                }, 500);
            }
            setTimeout(() => {
                // Alert.alert(`User tapped notification\n${JSON.stringify(notif)}`);
                Actions.login();
            }, 500);
        }
  
        if (Platform.OS === 'ios') {
            //optional
            //iOS requires developers
            //to call completionHandler to end notification process. If you do not call it your background remote notifications could be throttled, to read more about it see the above documentation link.
            //This library handles 
            // it for you automatically with default behavior (for remote notification, finish with NoData; for WillPresent, finish depend on "show_in_foreground"). However if you want to return different result, follow the following code to override
            //notif._notificationType is available for iOS platfrom
            switch (notif._notificationType) {
                case NotificationType.Remote:
                    notif.finish(RemoteNotificationResult.NewData); //other types available: RemoteNotificationResult.NewData, RemoteNotificationResult.ResultFailed
                    break;
                case NotificationType.NotificationResponse:
                    notif.finish();
                    break;
                case NotificationType.WillPresent:
                    notif.finish(WillPresentNotificationResult.All); //other types available: WillPresentNotificationResult.None
                    // this type of notificaiton will be called only when you are in foreground.
                    // if it is a remote notification, don't do any app logic here. Another notification callback will be triggered with type NotificationType.Remote
                    break;
                default:
                    break;
            }
        }
    });
}
