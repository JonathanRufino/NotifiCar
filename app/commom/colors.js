const Colors = {
    WHITE: '#FFF',
    BLACK: '#000',
    BLUE: '#1E88E5',
    RED: '#FF0000',
    GRAY: '#aeb2b7',
    LIGHT_GRAY: '#d6d8db',
    DARK_GRAY: '#686868',
    TRANSPARENT: 'transparent',
    GREEN: '#115E54',
    GREEN_DARK: '#114D44',
    RED_DARK: '#D23438',
    RED_MIDDLE: '#Ed3237',
    RED_LIGHT: '#Eb7575',
    FACEBOOK_BLUE: '#3B5998',
};

export { Colors };
