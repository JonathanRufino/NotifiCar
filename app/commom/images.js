const Images = {
    LOGIN_BACKGROUND: require('../assets/images/login_background.png'),
    LOGO_WHITE: require('../assets/images/logo-white.png'),
    LOGO_SPLASHSCREEN: require('../assets/images/logo-splashscreen.png'),
    ICON_ALARM: require('../assets/images/alarm.png'),
    ICON_LIGHTS: require('../assets/images/lights.png'),
    ICON_WARNING: require('../assets/images/warning.png'),
    ICON_WINDOW: require('../assets/images/window.png'),
    ICON_CAR: require('../assets/images/car.png'),
    ICON_ATTENTION: require('../assets/images/attention.png'),
    ICON_DOOR_OPEN: require('../assets/images/door_open.png'),
    ICON_CAR_BLOCKING: require('../assets/images/car_blocking.png'),
    ICON_FLAT_TIRE: require('../assets/images/flat_tire.png'),
    ICON_TRASH: require('../assets/images/icon-trash.png'),
};

export { Images };
