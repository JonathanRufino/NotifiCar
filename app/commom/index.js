export { Colors } from './colors';
export { Texts } from './texts';
export { Regexes } from './regexes';
