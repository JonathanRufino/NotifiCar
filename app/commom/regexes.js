const Regexes = {
    LICENSE_PLATE: /^[a-zA-Z]{3}-\d{4}$/,
};

export { Regexes };    
