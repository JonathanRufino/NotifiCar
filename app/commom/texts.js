const Texts = {
    Titles: {
        REGISTER_VEHICLE: 'Cadastrar Veículo',
    },
    Errors: {
        EMPTY_LICENSE_PLATE: 'Informe a placa do veículo',
        INVALID_LICENSE_PLATE: 'Informe uma placa válida',
    },
    Placeholders: {
        LICENSE_PLATE: 'AAA-0000',
    },
    Buttons: {
        REGISTER: 'Cadastrar',
    }
};

export { Texts };
