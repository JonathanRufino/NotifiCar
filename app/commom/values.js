import { Dimensions } from 'react-native';

const { width, height } = Dimensions.get('window');

const Values = {
    SCREEN_WIDTH: width,
    SCREEN_HEIGHT: height,
    EMPTY: 0,
    LICENSE_PLATE_MAX_LENGTH: 8,
    MAX_OCCURRENCE_ON_LIST_OF_OCCURRENCES: 10,
    FIRST: 0,
    SECOND: 1,
    THIRD: 2,
};

export { Values };
