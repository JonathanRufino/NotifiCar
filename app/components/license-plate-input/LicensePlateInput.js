import React, { Component } from 'react';
import { TextInputMask } from 'react-native-masked-text';
import PropTypes from 'prop-types';

import styles from './styles';
import { Values, Texts, Colors } from '../../commom';

class LicensePlateInput extends Component {
    state = {
        vehicle: '',
    };
    
    render() {
        return (
            <TextInputMask
                style={this.props.style}
                value={this.state.vehicle}
                onChangeText={text => {
                    this.setState({ vehicle: text });
                    this.props.onWrite(text);
                }}
                maxLength={this.props.maxLength}
                autoCapitalize={this.props.autoCapitalize}
                returnKeyType={this.props.returnKeyType}
                placeholder={this.props.placeholder}
                type='custom'
                options={{ mask: 'AAA-9999' }}
                underlineColorAndroid={Colors.TRANSPARENT}
            />
        );
    }
}

LicensePlateInput.propTypes = {
    style: PropTypes.any,
    maxLength: PropTypes.number,
    autoCapitalize: PropTypes.string,
    returnKeyType: PropTypes.string,
    placeholder: PropTypes.string,
    onWrite: PropTypes.func,
};

LicensePlateInput.defaultProps = {
    style: styles.inputField,
    maxLength: Values.LICENSE_PLATE_MAX_LENGTH,
    autoCapitalize: 'characters',
    returnKeyType: 'done',
    placeholder: Texts.Placeholders.LICENSE_PLATE,
    onWrite: () => console.log('Not implemented')
};

export default LicensePlateInput;
