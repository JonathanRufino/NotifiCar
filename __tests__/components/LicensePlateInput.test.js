import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import LicensePlateInput from '../../app/components/license-plate-input';
import { Images } from '../../app/commom';

describe('<LicensePlateInput />', () => {
    it('renders correctly', () => {
        const tree = renderer.create(
            <LicensePlateInput
                occurrenceType='Pneu Furado'
                image={Images.ICON_FLAT_TIRE}
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
