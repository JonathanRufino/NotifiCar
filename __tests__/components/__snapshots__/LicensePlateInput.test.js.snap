// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LicensePlateInput /> renders correctly 1`] = `
<TextInput
  allowFontScaling={true}
  autoCapitalize="characters"
  keyboardType="default"
  maxLength={8}
  onChangeText={[Function]}
  options={
    Object {
      "mask": "AAA-9999",
    }
  }
  placeholder="AAA-0000"
  returnKeyType="done"
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "#d6d8db",
      "borderBottomColor": "#000",
      "borderBottomWidth": 1,
      "borderColor": "#aeb2b7",
      "borderRadius": 10,
      "borderWidth": 5,
      "fontFamily": "Mandatory",
      "fontSize": 90,
      "marginVertical": 10,
      "textAlign": "center",
      "width": "80%",
    }
  }
  type="custom"
  underlineColorAndroid="transparent"
  value=""
/>
`;
